designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala

```diff
@@ -0,0 +1,16 @@
+package pl.touk.nussknacker.ui.config
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import scala.util.Random
+
+private object FingerprintUtils {
+  lazy val random = s"gen-${Random.alphanumeric.take(10).mkString}"
+}
+
+case class UsageStatisticsReportsConfig(enabled: Boolean, fingerprint: String = FingerprintUtils.random)
+
+object UsageStatisticsUrl {
+  def apply(fingerprint: String, version: String) =
+    s"https://stats.nussknacker.io/?fingerprint=${URLEncoder.encode(fingerprint, StandardCharsets.UTF_8)}&version=${URLEncoder.encode(version, StandardCharsets.UTF_8)}&timestamp=${System.currentTimeMillis()}"
```

Comment:
what may pros and cons of having this url hardcoded vs hardcoded via configuration ?

Reply:
configuration can be overriden :)